src/controller/CHIPDeviceController.h

```diff
@@ -167,6 +167,16 @@ struct CommissionerInitParams : public ControllerInitParams
     Credentials::DeviceAttestationVerifier * deviceAttestationVerifier = nullptr;
 };
 
+// Interface class for DeviceController methods that need to be mocked
+class IDeviceController
```

Comment:
Why is this here?  Nothing ever uses this type except for DeviceController inheriting from it, right?

This needs a better explanation of what this thing is for, at which point we can figure out a reasonable name for it.

Reply:
My bad, I introduced it for mocks in `TestCommissioningWindowOpener.cpp` and ended up overriding the `DeviceController`'s function instead. Just removed this abstract class and added virtual keyword to the function of interest (the one that needs to be overridden for tests): https://github.com/project-chip/connectedhomeip/pull/34209/commits/da28dd643e7419092f7a073ab2011fda251f2504

Reply:
Ah, this makes sense.  OK, great.